### ISSUE TITLE
collision leave

### ISSUE THREAD

**Issue by Shibohi** [YoYoGames/GameMaker-Bugs#4477]
### Is your feature request related to a problem?

i dont like the workflow of checking for collisions in the step event when i have to toggle a variable.
we cant use the collision events because we dont have an collision leave to reset the variable.

this feels odd and incomplete in the workflow and the code gets split in collision_events and step_event

### Describe the solution you'd like

it would be great if we had collision leave events similiar to the collision events.

or

if we could get a collision_all_enter / collision_all_leave were we can check the objects by our selfs.

### Describe alternatives you've considered

write every collision in the step event

**Comment by ParodyKnaveBob** [comment]
There is indeed something frustrating about wanting to (and being officially recommended to) offload as much computing from Step as possible and having Collision Events available but seemingly more often needing to avoid them.

I don't know if Collision End Event is the solution (since Collision Event is already taxing), despite our having had Mouse Enter and Mouse Leave since forever, but I suppose an alternative is for the user to set a variable in Collision that still has to get looked at in Step if not also Begin Step with yet another tracker variable.

**Comment by Totobal5** [comment]
Yes please is so weird to make triggers and zones using collisions and step
